packages/@react-aria/dnd/src/useDrop.ts

```diff
@@ -300,9 +312,10 @@ export function useDrop(options: DropOptions): DropResult {
     return allowedOperations[0];
   });
 
-  let {ref} = options;
+  let {ref, buttonRef} = options;
   useLayoutEffect(() => DragManager.registerDropTarget({
```

Comment:
buttonRef was added so that both the button/outer div could act as the drop target (instead of just one or the other) since in the case of dropzone, the user will want to be able to drop over any area of the dropzone rather than just the hidden button which is impossible to access when dropping with the mouse. focus still will be on the button during a drop which leads to the updates in the drag manager where focusElement is the hidden button. overall, there were some changes to the api but none that i believe should affect the other dnd components.

Reply:
do we need a ref to the outer div at all if the button is what gets focus during keyboard drop? does it work to just set `ref` to the button element rather than the div, and remove the extra `buttonRef`/`focusElement`?

Reply:
so the issue really isn't so much the keyboard drop as it is with dropping with a mouse. originally i had just set `ref` to the button which worked perfectly fine for the keyboard drop but not with the mouse. in this case, when you tried dropping with a mouse, an error would be thrown because the `e.target` would be the outer div, which was not considered valid drop target since only the button was created with useDrop. however, it would be impossible to target the button if you were only dragging with your mouse. that's why i ended up setting a ref to the outer div as well